e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/C_DiscoverEndpointsTestTheory.cs

```diff
@@ -0,0 +1,160 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace IIoTPlatform_E2E_Tests.Orchestrated {
+    using Newtonsoft.Json;
+    using RestSharp;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using TestExtensions;
+    using Xunit;
+    using Xunit.Abstractions;
+
+    [TestCaseOrderer(TestCaseOrderer.FullName, TestConstants.TestAssemblyName)]
+    [Collection("IIoT Multiple Nodes Test Collection")]
+    [Trait(TestConstants.TraitConstants.PublisherModeTraitName, TestConstants.TraitConstants.PublisherModeOrchestratedTraitValue)]
+    public class C_DiscoverEndpointsTestTheory {
+        private readonly ITestOutputHelper _output;
+        private readonly IIoTMultipleNodesTestContext _context;
+        private readonly CancellationTokenSource _cancellationToken;
```

Comment:
Please rename `_cancellationToken` to `_cancellationTokenSource` or `_cts`. `_cancellationToken` name is misleading.